crates/ruma-events/src/direct.rs

```diff
@@ -58,18 +115,20 @@ impl FromIterator<(OwnedUserId, Vec<OwnedRoomId>)> for DirectEventContent {
 mod tests {
     use std::collections::BTreeMap;
 
-    use ruma_common::{owned_room_id, owned_user_id};
+    use ruma_common::{owned_room_id, user_id, OwnedUserId};
+    // use ruma_macros::user_id;
     use serde_json::{from_value as from_json_value, json, to_value as to_json_value};
 
     use super::{DirectEvent, DirectEventContent};
+    use crate::direct::{DirectUserIdentifier, OwnedDirectUserIdentifier};
 
     #[test]
     fn serialization() {
         let mut content = DirectEventContent(BTreeMap::new());
-        let alice = owned_user_id!("@alice:ruma.io");
+        let alice = DirectUserIdentifier::from_borrowed("@alice:ruma.io");
```

Comment:
Let's keep using `(Owned)UserId` for user IDs in this test, this is probably the type users will use so we need to make sure the API plays well with that.

Reply:
Unfortunately I couldn't make that work with `into()` when trying. The compiler seems to be confused between `From<&UserId> for OwnedDirectUserIdentifier` and `From<&'a UserId> for &'a DirectUserIdentifier`. If I remove one (or the other ? I don't remember exactly) it works but it breaks other `into()` use cases.

It also work if I force the into impl with `Into<OwnedDirectUserIdentifier>::into()` but it's also ugly.

I would be quite happy if you find a solution for that :) I am still not really good at Rust.

Reply:
Yeah, that's normal, the compiler needs help sometimes. You are talking about L154 right? We would probably write it like that:

```rust
let direct_rooms = event.content.get(<&DirectUserIdentifier>::from(alice)).unwrap();
```